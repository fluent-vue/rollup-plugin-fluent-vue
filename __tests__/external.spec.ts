import { dirname, resolve } from 'path'
import { fileURLToPath } from 'url'
import { describe, expect, it } from 'vitest'

import rollupVue3 from 'rollup-plugin-vue'
import vue3base from '@vitejs/plugin-vue'
import compiler from '@vue/compiler-sfc'
import { createVuePlugin as vue2 } from 'vite-plugin-vue2'

import fluentPlugin from '../src'
import { testBundle } from './util/helpers'

const vue3 = () => vue3base({
  compiler,
})

const baseDir = dirname(fileURLToPath(import.meta.url))

describe.each(['development', 'production'])('external ftl file support mode:%s', (mode) => {
  it('works with vue 3', async() => {
    // Arrange
    // Act
    const code = await testBundle({
      mode,
      plugins: [
        vue3(),
        fluentPlugin({
          external: {
            baseDir: resolve(baseDir, 'fixtures'),
            ftlDir: resolve(baseDir, 'fixtures/ftl'),
            locales: ['en', 'da'],
          },
        }),
      ],
    }, '/fixtures/components/external.vue')

    // Assert
    expect(code).toMatchSnapshot()
  })

  it('works with vue 3 script setup', async() => {
    // Arrange
    // Act
    const code = await testBundle({
      mode,
      plugins: [
        vue3(),
        fluentPlugin({
          external: {
            baseDir: resolve(baseDir, 'fixtures'),
            ftlDir: resolve(baseDir, 'fixtures/ftl'),
            locales: ['en', 'da'],
          },
        }),
      ],
    }, '/fixtures/components/external.setup.vue')

    // Assert
    expect(code).toMatchSnapshot()
  })

  it('works with vue 3 rollup plugin', async() => {
    // Arrange
    // Act
    const code = await testBundle({
      mode,
      plugins: [
        rollupVue3(),
        fluentPlugin({
          external: {
            baseDir: resolve(baseDir, 'fixtures'),
            ftlDir: resolve(baseDir, 'fixtures/ftl'),
            locales: ['en', 'da'],
          },
        }),
      ],
    }, '/fixtures/components/external.vue')

    // Assert
    expect(code).toMatchSnapshot()
  })

  it('works with vue 2', async() => {
    // Arrange
    // Act
    const code = await testBundle({
      mode,
      plugins: [
        vue2(),
        fluentPlugin({
          external: {
            baseDir: resolve(baseDir, 'fixtures'),
            ftlDir: resolve(baseDir, 'fixtures/ftl'),
            locales: ['en', 'da'],
          },
        }),
      ],
    }, '/fixtures/components/external.vue')

    // Assert
    expect(code).toMatchSnapshot()
  })
})
