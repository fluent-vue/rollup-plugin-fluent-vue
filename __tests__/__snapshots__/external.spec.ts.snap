// Vitest Snapshot v1

exports[`external ftl file support > works with vue 2 1`] = `
"import { createHotContext as __vite__createHotContext } from \\"/@vite/client\\";import.meta.hot = __vite__createHotContext(\\"/fixtures/components/external.vue\\");import.meta.env = {\\"BASE_URL\\":\\"/\\",\\"MODE\\":\\"development\\",\\"DEV\\":true,\\"PROD\\":false,\\"SSR\\":false};import da_ftl from '/fixtures/ftl/da/components/external.vue.ftl?import';
import en_ftl from '/fixtures/ftl/en/components/external.vue.ftl?import';
const __vue2_script = {}
import { render as __vue2_render, staticRenderFns as __vue2_staticRenderFns } from \\"/fixtures/components/external.vue?vue&type=template&lang.js\\"
const __cssModules = {}

/* normalize component */
import __vue2_normalizer from \\"/@id/__x00__/vite/vueComponentNormalizer\\"
var __component__ = /*#__PURE__*/__vue2_normalizer(
  __vue2_script,
  __vue2_render,
  __vue2_staticRenderFns,
  false,
  __vue2_injectStyles,
  null,
  null,
  null
)

function __vue2_injectStyles (context) {
  for(let o in __cssModules){
    this[o] = __cssModules[o]
  }
}

__component__.options.fluent = __component__.options.fluent || {};
__component__.options.fluent['en'] = en_ftl
__component__.options.fluent['da'] = da_ftl
__component__.options.__file = \\"fixtures/components/external.vue\\"
/* hot reload */
import __VUE_HMR_RUNTIME__ from \\"/@id/__x00__/vite/vueHotReload\\"
import vue from \\"/@id/vue\\"
__VUE_HMR_RUNTIME__.install(vue)
if(!import.meta.env.SSR && __VUE_HMR_RUNTIME__.compatible){
  if (!__VUE_HMR_RUNTIME__.isRecorded(\\"4c0553ad\\")) {
    __VUE_HMR_RUNTIME__.createRecord(\\"4c0553ad\\", __component__.options)
  }
   import.meta.hot.accept((update) => {
      __VUE_HMR_RUNTIME__.reload(\\"4c0553ad\\", update.default)
   })
   import.meta.hot.accept(\\"/fixtures/components/external.vue?vue&type=template&lang.js\\", (update) => {
      __VUE_HMR_RUNTIME__.rerender(\\"4c0553ad\\", update)
   })
}
export default /*#__PURE__*/(function () { return __component__.exports })()"
`;

exports[`external ftl file support > works with vue 3 1`] = `
"import { createHotContext as __vite__createHotContext } from \\"/@vite/client\\";import.meta.hot = __vite__createHotContext(\\"/fixtures/components/external.vue\\");import da_ftl from '/fixtures/ftl/da/components/external.vue.ftl?import';
import en_ftl from '/fixtures/ftl/en/components/external.vue.ftl?import';
const _sfc_main = {}
import { toDisplayString as _toDisplayString, createElementVNode as _createElementVNode, openBlock as _openBlock, createElementBlock as _createElementBlock } from \\"/@id/vue\\"

function _sfc_render(_ctx, _cache) {
  return (_openBlock(), _createElementBlock(\\"div\\", null, [
    _createElementVNode(\\"div\\", null, _toDisplayString(_ctx.$t('hello-user', { userName: _ctx.userName })), 1 /* TEXT */),
    _createElementVNode(\\"div\\", null, _toDisplayString(_ctx.$t('shared-photos', { userName: _ctx.userName, photoCount: _ctx.photoCount, userGender: _ctx.userGender })), 1 /* TEXT */)
  ]))
}


_sfc_main.fluent = _sfc_main.fluent || {};
_sfc_main.fluent['en'] = en_ftl
_sfc_main.fluent['da'] = da_ftl
_sfc_main.__hmrId = \\"4c0553ad\\"
typeof __VUE_HMR_RUNTIME__ !== 'undefined' && __VUE_HMR_RUNTIME__.createRecord(_sfc_main.__hmrId, _sfc_main)
import.meta.hot.accept(({ default: updated, _rerender_only }) => {
  if (_rerender_only) {
    __VUE_HMR_RUNTIME__.rerender(updated.__hmrId, updated.render)
  } else {
    __VUE_HMR_RUNTIME__.reload(updated.__hmrId, updated)
  }
})
import _export_sfc from '/@id/plugin-vue:export-helper'
export default /*#__PURE__*/_export_sfc(_sfc_main, [['render',_sfc_render],['__file',\\"/home/demivan/projects/fluent/rollup-plugin-fluent-vue/__tests__/fixtures/components/external.vue\\"]])"
`;

exports[`external ftl file support > works with vue 3 rollup plugin 1`] = `
"import da_ftl from '/fixtures/ftl/da/components/external.vue.ftl?import';
import en_ftl from '/fixtures/ftl/en/components/external.vue.ftl?import';
const script = {}
import { render } from \\"/fixtures/components/external.vue?vue&type=template&id=8d0ca2ca&lang.js\\"


script.render = render
script.fluent = script.fluent || {};
script.fluent['en'] = en_ftl
script.fluent['da'] = da_ftl
script.__file = \\"__tests__/fixtures/components/external.vue\\"
export default script"
`;

exports[`external ftl file support > works with vue 3 script setup 1`] = `
"import { createHotContext as __vite__createHotContext } from \\"/@vite/client\\";import.meta.hot = __vite__createHotContext(\\"/fixtures/components/external.setup.vue\\");import en_ftl from '/fixtures/ftl/en/components/external.setup.vue.ftl?import';
const _sfc_main = {
  setup(__props, { expose }) {
  expose();

const test = 'test'

const __returned__ = { test }
Object.defineProperty(__returned__, '__isScriptSetup', { enumerable: false, value: true })
return __returned__
}

}
import { toDisplayString as _toDisplayString, createElementVNode as _createElementVNode, openBlock as _openBlock, createElementBlock as _createElementBlock } from \\"/@id/vue\\"

function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return (_openBlock(), _createElementBlock(\\"div\\", null, [
    _createElementVNode(\\"div\\", null, _toDisplayString(_ctx.$t('hello-user', { userName: _ctx.userName })), 1 /* TEXT */),
    _createElementVNode(\\"div\\", null, _toDisplayString(_ctx.$t('shared-photos', { userName: _ctx.userName, photoCount: _ctx.photoCount, userGender: _ctx.userGender })), 1 /* TEXT */)
  ]))
}


_sfc_main.fluent = _sfc_main.fluent || {};
_sfc_main.fluent['en'] = en_ftl
_sfc_main.__hmrId = \\"f4d31108\\"
typeof __VUE_HMR_RUNTIME__ !== 'undefined' && __VUE_HMR_RUNTIME__.createRecord(_sfc_main.__hmrId, _sfc_main)
import.meta.hot.accept(({ default: updated, _rerender_only }) => {
  if (_rerender_only) {
    __VUE_HMR_RUNTIME__.rerender(updated.__hmrId, updated.render)
  } else {
    __VUE_HMR_RUNTIME__.reload(updated.__hmrId, updated)
  }
})
import _export_sfc from '/@id/plugin-vue:export-helper'
export default /*#__PURE__*/_export_sfc(_sfc_main, [['render',_sfc_render],['__file',\\"/home/demivan/projects/fluent/rollup-plugin-fluent-vue/__tests__/fixtures/components/external.setup.vue\\"]])"
`;
